lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -115,7 +115,7 @@ namespace {
 
       // In type space, we reuse HEAD to help us print meaningful name, e.g.,
       // tuple element name in fixits.
-      Identifier Head;
+      DeclName Head;
```

Comment:
This isn't your bug, but "Head" no longer makes sense under these changes.  Perhaps `Name` or `Spine` (definitely rooting more for the former than the latter).